File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int cents;

int get_cents(void)
{
    // TODO
    do
    {
        cents = get_int("Number of cents? ");
    }
    while (!(cents > 0));

    printf("%i", )
}

int calculate_quarters(int cents)
{
    // TODO
    cents / 25 = int quart;
    printf("%i", quart);

}

int calculate_dimes(int cents)
{
    // TODO
    cents / 10 = int dim;
    printf("%i", dim);

}

int calculate_nickels(int cents)
{
    // TODO
    cents / 5 = int nick;
    printf("%i", nick);

}

int calculate_pennies(int cents)
{
    // TODO
    cents / 1 = int pen;
    printf("%i", pen);
}

File: sentimental-cash/cash.py
# TODO
from cs50 import get_float

#Define the funcctions
def calculate_quarters(dollars):
    return int((dollars * 100) // 25)

def calculate_dimes(dollars):
    return int((dollars * 100) // 10)

def calculate_nickels(dollars):
    return int((dollars * 100) // 5)

def calculate_pennies(dollars):
    return int((dollars * 100) // 1)

#Ask how many dollars the customer is owed
while True:
    dollars = get_float("Enter change owed ")
    if dollars < 0:
        print("Error, invalid amount")
        continue
    else:
        break

#Calculate the number of quarters to give the customer
quart = int(calculate_quarters(dollars))
dollars = dollars - quart * 25

#Calculate the number of dimes to give the customer
dimes = int(calculate_dimes(dollars))
dollars = dollars - dimes * 10

#Calculate the number of nickels to give the customer
nickels = int(calculate_nickels(dollars))
dollars = dollars - nickels * 5

#Calculate the number of pennies to give the customer
pennies = int(calculate_pennies(dollars))
dollars = dollars - pennies * 1

#The minimum number of coins
coins = int(quart + dimes + nickels + pennies)

#Print total number of coins to give the customer
print(coins)


